### PR TITLE
[WIP] Re-Introduce Helix View Aggregator

### PR DESCRIPTION
**Pull request by desaikomal**
This is WIP and this change only ensures compilation of helix-view-aggregator

### Issues

- [ ] My PR addresses the following Helix issues and references them in the PR description:

Any of the following keywords can be used: close, closes, closed, fix, fixes, fixed, resolve, resolves, resolved)

### Description

- [ ] Here are some details about my PR, including screenshots of any UI changes:

(Write a concise description including what, why, how)

### Tests

- [ ] The following tests are written for this issue:

(List the names of added unit/integration tests)

- The following is the result of the "mvn test" command on the appropriate module:

(If CI test fails due to known issue, please specify the issue and test PR locally. Then copy & paste the result of "mvn test" to here.)

### Changes that Break Backward Compatibility (Optional)

- My PR contains changes that break backward compatibility or previous assumptions for certain methods or API. They include:

(Consider including all behavior changes for public methods or API. Also include these changes in merge description so that other developers are aware of these changes. This allows them to make relevant code changes in feature branches accounting for the new method/API behavior.)

### Documentation (Optional)

- In case of new functionality, my PR adds documentation in the following wiki page:

(Link the GitHub wiki you added)

### Commits

- My commits all reference appropriate Apache Helix GitHub issues in their subject lines. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Code Quality

- My diff has been formatted using helix-style.xml 
(helix-style-intellij.xml if IntelliJ IDE is used)

### CODE REVIEW

File: helix-view-aggregator/pom.xml [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+  </properties>
+
+  <dependencies>
+    <dependency>
+      <groupId>io.dropwizard.metrics</groupId>
+      <artifactId>metrics-core</artifactId>
+      <version>3.2.3</version>
+    </dependency>
+    <dependency>
+      <groupId>org.slf4j</groupId>
```
Comment (narendly) [inline helix-view-aggregator/pom.xml:50]: Is this logger version new? Let's make sure we don't use versions that are known to have vulnerabilities.

File: helix-core/src/main/java/org/apache/helix/model/ClusterConfig.java [hunk 2]
```diff
@@ -82,6 +84,10 @@
     // error exceeds this limitation
     DISABLED_INSTANCES,
 
+    VIEW_CLUSTER, // Set to "true" to indicate this is a view cluster
+    VIEW_CLUSTER_SOURCES, // Map field, key is the name of source cluster, value is
+    // ViewClusterSourceConfig JSON string
+    VIEW_CLUSTER_REFRESH_PERIOD, // In second
```
[the remark this answers is not in the document]
Comment (desaikomal) [reply helix-core/src/main/java/org/apache/helix/model/ClusterConfig.java:90]: This was lift-n-shift from old existing code. Can I defer this improvement for next change?

File: helix-view-aggregator/error [hunk 3]
```diff
@@ -0,0 +1,103 @@
+[INFO] Scanning for projects...
```
[the remark this answers is not in the document]
Comment (desaikomal) [reply helix-view-aggregator/error:1]: this was due to addition of the whole directory. I will clean up.

File: helix-view-aggregator/pom.xml [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+  <modelVersion>4.0.0</modelVersion>
+
+  <artifactId>helix-view-aggregator</artifactId>
+  <packaging>bundle</packaging>
+  <name>Apache Helix :: View Aggregator</name>
+
+  <properties>
+    <osgi.import>
+      org.apache.helix*,
+      org.codehaus.jackson*,
```
[the remark this answers is not in the document]
Comment (desaikomal) [reply helix-view-aggregator/pom.xml:35]: i will make the change as i had to fix jackson dependencies in class to use fasterxml already.

File: helix-view-aggregator/pom.xml (hunk 1 above)
Comment (desaikomal) [reply helix-view-aggregator/pom.xml:50]: this is what helix-controller is using.. will change later for all of our components.

**Comment by junkaixue** [comment]
It seemed to be pure copy of the code? Is this possible we have historical commits ported from the branch?

**Comment by narendly** [comment]
> > I don't know how to get commit history for view-aggregator. Let me do some more research on how to get that and update the PR

If you do a format-patch, then you can preserve the commit history.

File: helix-view-aggregator/view-aggregator.txt [hunk 5]
```diff
@@ -0,0 +1,48 @@
+Date:   Thu May 23 18:00:38 2019 -0700
```
Comment (narendly) [inline helix-view-aggregator/view-aggregator.txt:1]: This isn't how format patch works..

**Pull request closed without merging**